scheduler/src/cook/rest/api.clj

```diff
@@ -362,7 +362,7 @@
       (dissoc (s/optional-key :group))
       (dissoc (s/optional-key :status))
       (merge {:framework-id (s/maybe s/Str)
-              :retries-remaining NonNegInt
+              :retries-remaining s/Int
```

Comment:
There's a bug in the retries remaining logic that causes this number to sometimes be negative. It's not clear yet where/what the bug is, but when it happens, this schema is causing us to not be able to query the job through the API, which is frustrating for users and makes it very difficult to track down the bug. So, while we're not yet fixing the bug, we're making it easier to troubleshoot by making the response schema less strict.

Reply:
Let's create a separate PR for this. Also, we should add a TODO and surrounding logs to detect when the retries remaining becomes negative.

Reply:
Created #1455 for this.